src/components/activity-list/ActivityList.tsx

```diff
@@ -0,0 +1,133 @@
+import { TransactionSections, TransactionItemForSectionList } from '@/helpers/buildTransactionsSectionsSelector';
+
+const sx = StyleSheet.create({
+  sectionHeader: {
+    paddingVertical: 18,
+  },
+});
+
+const ActivityListHeaderHeight = 42;
+const TRANSACTION_COIN_ROW_VERTICAL_PADDING = 7;
+
+const getItemLayout = sectionListGetItemLayout({
+  getItemHeight: () => CoinRowHeight + TRANSACTION_COIN_ROW_VERTICAL_PADDING * 2,
+  getSectionHeaderHeight: () => ActivityListHeaderHeight,
+});
+
+const keyExtractor = (data: TransactionSections['data'][number]) => {
+  if ('hash' in data) {
+    return (data.hash || data.timestamp ? data.timestamp?.toString() : performance.now().toString()) ?? performance.now().toString();
+  }
+  return (
+    (data.displayDetails?.timestampInMs ? data.displayDetails.timestampInMs.toString() : performance.now().toString()) ??
+    performance.now().toString()
+  );
+};
+
+const renderSectionHeader = ({ section, colors }: { section: TransactionSections; colors: ThemeContextProps['colors'] }) => {
+  return (
+    <View style={[sx.sectionHeader, { backgroundColor: colors.white }]}>
+      <ActivityListHeader {...section} />
+    </View>
+  );
+};
+
+const LoadingSpinner = android ? Spinner : ActivityIndicator;
+
+const FooterWrapper = styled(ButtonPressAnimation)({
+  alignItems: 'center',
+  height: 40,
+  justifyContent: 'center',
+  paddingBottom: 10,
+  width: '100%',
+});
+
+function ListFooterComponent({ label, onPress }: { label: string; onPress: () => void }) {
+  const [isLoading, setIsLoading] = useState(false);
+  const { colors } = useTheme();
+
+  useEffect(() => {
+    if (isLoading) {
+      onPress();
+      setIsLoading(false);
+    }
+  }, [isLoading, setIsLoading, onPress]);
+  const onPressWrapper = () => {
+    setIsLoading(true);
+  };
+  return (
+    <FooterWrapper onPress={onPressWrapper}>
+      {isLoading ? (
+        <LoadingSpinner />
+      ) : (
+        <Text align="center" color={colors.alpha(colors.blueGreyDark, 0.3)} lineHeight="loose" size="smedium" weight="bold">
+          {label}
+        </Text>
+      )}
+    </FooterWrapper>
+  );
+}
+
+const ActivityList = () => {
+  const { accountAddress, nativeCurrency } = useAccountSettings();
+
+  const { setScrollToTopRef } = useSectionListScrollToTopContext();
+  const { sections, nextPage, transactionsCount, remainingItemsLabel } = useAccountTransactions();
+  const pendingTransactions = usePendingTransactionsStore(state => state.pendingTransactions[accountAddress] || []);
+
+  const { colors } = useTheme();
+  const renderSectionHeaderWithTheme = useCallback(
+    ({ section }: { section: TransactionSections }) => renderSectionHeader({ colors, section }),
+    [colors]
+  );
+
+  const handleScrollToTopRef = (ref: SectionList<TransactionItemForSectionList, TransactionSections> | null) => {
+    if (!ref) return;
+    // @ts-expect-error - no idea why this is not working
```

Comment:
welcoming ideas on how to get this to work if anyone has one

Reply:
```ts
  const { setScrollToTopRef } = useSectionListScrollToTopContext<TransactionItemForSectionList, TransactionSections>();
```

Reply:
THANK YOU